src/tx/partition.rs

```diff
@@ -22,12 +19,10 @@ impl GarbageCollection for TransactionalPartitionHandle {
     }
 
     fn gc_with_space_amp_target(&self, factor: f32) -> crate::Result<u64> {
-        let _lock = self.tx_lock.lock().expect("lock is poisoned");
```

Comment:
Not sure if I can really test this for single writer, but both single writer and SSI tx's should work without an exclusive lock, because:

- the active memtable is locked for GC anyway, **so we already have an exclusive lock**, so writes are serializable (in fact serialized): https://github.com/fjall-rs/lsm-tree/blob/872461146969811a21f307cee8cc5ea56de156e8/src/blob_tree/mod.rs#L125, so while the GC runs, no Tx can commit
- a GC shouldn't change the tree to an outsider's (a running SSI Tx) perspective, so when the SSI commits, it does not conflict with the GC because the GC never changed anything logically
- the Tx should have a higher seqno than the GC, I have added a unit test that an SSI Tx is not shadowed by a concurrent GC - this is guaranteed by the Tx calling `Batch::commit`, which calls `seqno::next()`